crates/core/src/action.rs

```diff
@@ -163,7 +164,8 @@ pub enum Action {
 
     /// Makes the network module send a direct (node-to-node) message
     /// to the address given in [DirectMessageData](struct.DirectMessageData.html)
-    SendDirectMessage(DirectMessageData),
+    /// Includes the timeout information: system time of dispatch and duration until it timeouts.
+    SendDirectMessage((DirectMessageData, Option<(SystemTime, Duration)>)),
```

Comment:
@lucksus this pattern keeps showing up, maybe we can wrap this up in a struct with some methods rather than having these complex tuples everywhere